internal/driver/glfw/window_wayland.go

```diff
@@ -2,7 +2,30 @@
 
 package glfw
 
+import (
+	"unsafe"
+
+	"fyne.io/fyne/v2/driver"
+)
+
 // GetWindowHandle returns the window handle. Only implemented for X11 currently.
 func (w *window) GetWindowHandle() string {
 	return "" // TODO: Find a way to get the Wayland handle for xdg_foreign protocol. Return "wayland:{id}".
 }
+
+// assert we are implementing driver.NativeWindow
+var _ driver.NativeWindow = (*window)(nil)
+
+func (w *window) RunNative(f func(any) error) error {
+	var err error
+	done := make(chan struct{})
+	runOnMain(func() {
+		err = f(driver.WaylandWindowContext{
+			WaylandSurface: uintptr(unsafe.Pointer(w.view().GetWaylandWindow())),
+			EGLSurface:     uintptr(unsafe.Pointer(w.view().GetEGLSurface())),
```

Comment:
Do we need convert to `unsafe.Pointer()` here? I see you are not doing it for the `darwin` target.

Reply:
Yes, as the type is not directly convertible to uintptr